src/main.c

```diff
@@ -814,7 +817,32 @@ int main(int argc, char *argv[]) {
 #endif
 
   // Process command line arguments
-  while((ch = getopt(argc, argv, "?Ab:B:c:C:DeE:Fi:l:nNp:OP:qrtT:U:vVx:")) != -1) {
+  struct option longopts[] = {
+    {"help",       no_argument,       NULL, '?'},
+    {"baud",       required_argument, NULL, 'b'},
+    {"bitclock",   required_argument, NULL, 'B'},
+    {"programmer", required_argument, NULL, 'c'},
+    {"config",     required_argument, NULL, 'C'},
+    {"noerase",    no_argument,       NULL, 'D'},
+    {"erase",      no_argument,       NULL, 'e'},
+    {"logfile",    required_argument, NULL, 'l'},
+    {"test",       no_argument,       NULL, 'n'},
+    {"noconfig",   no_argument,       NULL, 'N'},
+    {"part",       required_argument, NULL, 'p'},
+    {"chip",       required_argument, NULL, 'p'},
+    {"port",       required_argument, NULL, 'P'},
+    {"quiet",      no_argument,       NULL, 'q'},
+    {"reconnect",  no_argument,       NULL, 'r'},
+    {"terminal",   no_argument,       NULL, 't'},
+    {"tty",        no_argument,       NULL, 't'},
+    {"memory",     required_argument, NULL, 'U'},
```

Comment:
`--memory` is a good long option name b/c it puts the focus correctly: the operations `r`, `w` and `v` all refer to memory and, crucially, not the corresponding file; originally `-U` got its name from the generic update operations it might describe, so I would also allow the mnemonic `--update` or, alternatively, `--update-memory`. I have an ever so slight  preference for the latter: personally, I think long options are good in shell scripts to improve readability, so I am less worried about the length of long options. If we go with the latter, we wouldn't need `--memory`.

Reply:
I would argue that `--memory` would be sufficient. The help test will get awfully long if we have tons of long options that to the same thing. And the -U flag doesn't necessary _updates_ the memory.

IMO, `--memory flash:r:myfile.hex` makes way more sense than `--update-memory flash:r:myfile.hex`.